### PR TITLE
Refactors the way we work with compute SKUs

### PR DESCRIPTION
**Pull request by m1kola**
### Which issue this PR addresses:

Part of work item [№9586080](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/9586080/).

### What this PR does / why we need it:

This is a prerequisite for Server-Side Encryption and Encryption at host validations PR https://github.com/Azure/ARO-RP/pull/1595.

### Test plan for issue:

Existing unit tests and e2e tests should cover this refactoring.

### Is there any documentation that needs to be updated for this PR?

No, just refactoring.

### CODE REVIEW

File: pkg/util/computeskus/computeskus.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+func Zones(sku *mgmtcompute.ResourceSku) []string {
+	if sku.LocationInfo == nil || len(*sku.LocationInfo) == 0 {
+		return nil
+	}
+
+	return *(*sku.LocationInfo)[0].Zones
+}
+
+// Capabilities checks whether given resource SKU has specific capability
+func HasCapability(sku *mgmtcompute.ResourceSku, capabilityName string) bool {
```
Comment (m1kola) [inline pkg/util/computeskus/computeskus.go:20]: Note: we are not using this one anywhere, but will need it in https://github.com/Azure/ARO-RP/pull/1595

File: pkg/util/computeskus/computeskus.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-01/compute"
+)
+
+// Zones returns zone information for the resource SKU
+func Zones(sku *mgmtcompute.ResourceSku) []string {
+	if sku.LocationInfo == nil || len(*sku.LocationInfo) == 0 {
+		return nil
+	}
+
+	return *(*sku.LocationInfo)[0].Zones
```
Comment (gvanderpotte) [inline pkg/util/computeskus/computeskus.go:16]: I might be paranoid, but should not we test the "nillity" of `(*sku.LocationInfo)[0].Zones` before dereferencing it ?

**Review by gvanderpotte — COMMENTED** [review]
one comment, but looks good to me otherwise

File: pkg/util/computeskus/computeskus.go (hunk 2 above)
Comment (m1kola) [reply pkg/util/computeskus/computeskus.go:16]: `len(*sku.LocationInfo) == 0` above makes sure that `*sku.LocationInfo` doesn't point to `nil` (`len()` can take `nil`) and not empty, so there should be no `nil` dereference error.

As for checking whether `*(*sku.LocationInfo)[0].Zones` is `nil` or not - it should be a concern of the calling code.

Or did I misunderstood something?

Comment (m1kola) [reply pkg/util/computeskus/computeskus.go:16]: For visibility. Had a chat with Ghislain about it. Here is what I missed: `*sku.LocationInfo[0].Zones` is a `*[]string`, There will be dereferencing error in `*(*sku.LocationInfo)[0].Zones`, if it is `nil`.

I'll add an extra check and add a unit test case for it shortly.

Comment (m1kola) [reply pkg/util/computeskus/computeskus.go:18]: Updated.

**Review by gvanderpotte — APPROVED** [review]
LGTM

File: pkg/env/prod.go [hunk 3]
```diff
@@ -128,7 +128,8 @@ func newProd(ctx context.Context, log *logrus.Entry) (*prod, error) {
 
 	p.serviceKeyvault = keyvault.NewManager(msiKVAuthorizer, serviceKeyvaultURI)
 
-	err = p.populateZones(ctx, msiAuthorizer)
+	resourceSkusClient := compute.NewResourceSkusClient(p.Environment(), p.SubscriptionID(), msiAuthorizer)
+	err = p.populateVMSkus(ctx, resourceSkusClient)
```
Comment (mjudeikis) [inline pkg/env/prod.go:132]: Any reason this logic changed in the way that:
before: client initiated inside `populateZones` and authorized passed in.
now: client initiated outside an client passed in?

**Review by mjudeikis — APPROVED** [review]
Small nit question, but overall ok

File: pkg/env/prod.go (hunk 3 above)
Comment (m1kola) [reply pkg/env/prod.go:132]: @mjudeikis the only reasons is to make `populateZones` (now `populateVMSkus`) more unit testable. I did not add any extra coverage as part of this PR

**Pull request merged**